scalalib/src/mill/scalalib/publish/LocalIvyPublisher.scala

```diff
@@ -13,46 +13,99 @@ class LocalIvyPublisher(localIvyRepo: os.Path) {
       ivy: os.Path,
       artifact: Artifact,
       extras: Seq[PublishInfo]
-  )(implicit ctx: Ctx.Log): Unit = publishLocal(jar, sourcesJar, docJar, pom, ivy, artifact, extras)
+  )(implicit ctx: Ctx.Log): Unit = {
+    val mainArtifacts = Seq(
+      LocalIvyPublisher.jarPublishInfo(jar),
+      LocalIvyPublisher.sourcesJarPublishInfo(sourcesJar),
+      LocalIvyPublisher.docJarPublishInfo(docJar)
+    )
+    publishLocal(pom, Right(ivy), artifact, mainArtifacts ++ extras)
+  }
 
+  /**
+   * Publishes a module locally
+   *
+   * @param pom The POM of this module
+   * @param ivy If right, the path to the ivy.xml file of this module; if left, its content as a String
+   * @param artifact Coordinates of this module
+   * @param extras Files to publish in this module
+   * @param ctx
+   * @return The files created or written to when publishing locally this module
+   */
   def publishLocal(
-      jar: os.Path,
-      sourcesJar: os.Path,
-      docJar: os.Path,
```

Comment:
We should rename `extras` to `publishInfos` or `publishData`, since this is no longer "extra" but "the" data.